### PR TITLE
Modify package to target netstandard1.5

### PR DESCRIPTION
**Pull request by pakrym**
Update target framework, folder names, package versions,.
Also updates nuget.exe to latest version from nuget.org because current version does not understand `netstandard*`.

@shirhatti @dnduffy @SergeyKanzhelev

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
Mention the bug to associate with: https://github.com/Microsoft/ApplicationInsights-dotnet/issues/335

**Comment by pakrym** [comment]
Actually `Profile7` is `netstandard1.1` based on https://github.com/davidfowl/NetStandard/blob/master/PclToNetStandard.md so moving to `netstandard1.1`

**Comment by SergeyKanzhelev** [comment]
I think our desire is to switch to the 1.5 and change compilation from PCL to the regular .NET Core. This way we will not need most of special logic around missing APIs.

**Comment by SergeyKanzhelev** [comment]
So let's keep 1.5 to avoid possible impact.

File: src/Core/NuGet/Core/Package.nuspec [hunk 1]
```diff
@@ -28,34 +28,34 @@
...
-        <dependency id="System.Reflection.Extensions" version="4.0.0" />
-        <dependency id="System.Runtime" version="4.0.20" />
-        <dependency id="System.Runtime.Extensions" version="4.0.10" />
-        <dependency id="System.Text.Encoding" version="4.0.10" />
-        <dependency id="System.Text.Encoding.Extensions" version="4.0.10" />
-        <dependency id="System.Text.RegularExpressions" version="4.0.10" />
-        <dependency id="System.Threading" version="4.0.10" />
-        <dependency id="System.Threading.Tasks" version="4.0.10" />
-        <dependency id="System.Xml.XDocument" version="4.0.10" />
+      <group targetFramework="netstandard1.5">
```
Comment (dnduffy) [inline src/Core/NuGet/Core/Package.nuspec]: Any reason not to go to 1.6?

Comment (pakrym) [inline src/Core/NuGet/Core/Package.nuspec]: In general we want to go as low as possible, so more libraries can depend on this one.

**Comment by pakrym** [comment]
@dnduffy when can we expect package appearing on feed?

**Comment by dnduffy** [comment]
@pakrym Excellent question, do you just need a private beta or something more official?

**Comment by pakrym** [comment]
I was thinking about `myget.org/F/applicationinsights` so I can start consuming it from my aspnetcore branch.

**Comment by dnduffy** [comment]
Right, I'll put a build up on MyGet for the 2.2.0 train after I finish reviewing the other PR.

**Comment by dnduffy** [comment]
I communicated this by IM earlier; here is the package:
https://www.myget.org/feed/applicationinsights/package/nuget/Microsoft.ApplicationInsights/2.2.0-beta5

**Comment by tinchou** [comment]
Does this PR make it compatible with `netcoreapp1.1`? I'm waiting just for this project to update to ASP.NET Core 1.1 :)

**Comment by dnduffy** [comment]
@tinchou The .NET Core SDK 2.0.0 references the base SDK (this) 2.2.0 stable and it works with netcoreapp1.0 and 1.1.

**Pull request merged**